IdentityCore/src/controllers/broker/mac/MSIDXpcSilentTokenRequestController.m

```diff
@@ -28,20 +28,44 @@
 #import "MSIDLogger+Internal.h"
 #import "MSIDXpcProviderCache.h"
 
+@interface MSIDXpcSilentTokenRequestController ()
+
+// shouldSkipXpcRequest is used when the fallback is not needed for non ASAuthorizationErrorDomain or MSIDErrorSSOExtensionUnexpectedError
+@property (nonatomic) BOOL shouldSkipAcquireToken;
+
+@end
+
 @implementation MSIDXpcSilentTokenRequestController
 
 - (void)acquireToken:(MSIDRequestCompletionBlock)completionBlock
 {
-    MSID_LOG_WITH_CTX(MSIDLogLevelInfo, self.requestParameters, @"Beginning silent broker xpc flow.");
-    MSIDRequestCompletionBlock completionBlockWrapper = ^(MSIDTokenResult *result, NSError *error)
+    MSID_LOG_WITH_CTX(MSIDLogLevelInfo, self.requestParameters, @"Beginning silent broker xpc flow, should use Xpc flow to acquire token: %@", @(!self.shouldSkipAcquireToken));
+    if (!self.shouldSkipAcquireToken)
     {
-        MSID_LOG_WITH_CTX(MSIDLogLevelInfo, self.requestParameters, @"Silent broker xpc flow finished. Result %@, error: %ld error domain: %@, shouldFallBack: %@", _PII_NULLIFY(result), (long)error.code, error.domain, @(self.fallbackController != nil));
-        completionBlock(result, error);
-    };
-    
-    __auto_type request = [self.tokenRequestProvider silentXpcTokenRequestWithParameters:self.requestParameters
-                                                                            forceRefresh:self.forceRefresh];
-    [self acquireTokenWithRequest:request completionBlock:completionBlockWrapper];
+        MSIDRequestCompletionBlock completionBlockWrapper = ^(MSIDTokenResult *result, NSError *error)
+        {
+            MSID_LOG_WITH_CTX(MSIDLogLevelInfo, self.requestParameters, @"Silent broker xpc flow finished. Result %@, error: %ld error domain: %@, shouldFallBack: %@", _PII_NULLIFY(result), (long)error.code, error.domain, @(self.fallbackController != nil));
+            completionBlock(result, error);
+        };
+        
+        __auto_type request = [self.tokenRequestProvider silentXpcTokenRequestWithParameters:self.requestParameters
+                                                                                forceRefresh:self.forceRefresh];
+        [self acquireTokenWithRequest:request completionBlock:completionBlockWrapper];
+    }
+    else
+    {
+        // self.fallbackController cannot be nil here as it has been validated from caller
```

Comment:
Is this comment inaccurate? If we know fallbackController cannot be nil here, why do a nil check and log in the code? Please remove the comment as it isn't accurate according to the code following it